lessons/07-mcp/README.md

```diff
@@ -0,0 +1,482 @@
+# MCP, Model Context Protocol
+
+In this chapter you will learn:
+
+- How to use the Model Context Protocol, MCP to split up your server capabilities in a server and a client
+- Create tools, resources and prompts on the server.
+- Consume the server using both the inspector tool and a written client.
+
+## Setup
+
+If you haven't already, set up your development environment. Here's how you can do it: [Setup your environment](/docs/setup/README.md).
+
+## Related resources
+
+[![Watch a short video about MCP](https://img.youtube.com/vi/YRfOiB0Im64/0.jpg)](https://www.youtube.com/watch?v=YRfOiB0Im64)
+
+_This video explains Model Context Protocol._
+
+*🎥 Click on the image above to watch a short video about MCP*
+
+## Narrative: Scipio Africanus
+
+> [!NOTE] 
+> _Our story so far. You are a mechanic from 1860s London. You were working on your automaton and received a letter from Charles Babbage that ended up taking you to a library where you picked up a time travel device. Throughout your travels in time you've ended up in Florence, where you met Leonardo Da Vinci. You also went to the Aztec empire and this is where the story continues._
+>
+> See [Lesson 1](../01-intro-to-genai/README.md) if you want to catch up with the story from the beginning. 
+
+> [!NOTE] 
+> While we recommend going through the story (it's fun!), [click here](#interact-with-scipio) if you'd prefer to jump straight to the technical content.
+
+_Our heroes had just learned tools and using tools via tool calling inspired by their meeting with Amelia Earhart and is now returning back to Ada Lovelace's mansion to discuss their new found knowledge. A flash of light envelops our heroes as the time beetle transports them back to Aaa's present. They find themselves in the basement of Ada Lovelace's mansion, Charles Babbage is not present, but Ada is there waiting for them. She is excited to hear about their journey and the knowledge they've gained._
+
+**Ada**: "Welcome back! I trust your journey was enlightening."
+
+**You**: "It was! meeting with mrs Earhart was quite the experience. She taught us about the importance of tools and how to use them effectively." 
+
+**Ada**: "There are few people who understand the importance of tools better than Amelia. I trust you've also upgraded her time beetle?"
```

Comment:
We seem to be using "Time Beetle" in the other chapters so I'd uppercase it. Kind of like a mechanical version of a human. :-)